agent/telemetry.go

```diff
@@ -0,0 +1,126 @@
+package agent
+
+import (
+	"fmt"
+	"time"
+
+	metrics "github.com/armon/go-metrics"
+	"github.com/armon/go-metrics/circonus"
+	"github.com/armon/go-metrics/datadog"
+	"github.com/armon/go-metrics/prometheus"
+	"github.com/hashicorp/nomad-autoscaler/agent/config"
+)
+
+// setupTelemetry is used to setup the telemetry sub-systems and returns the
+// in-memory sink to be used in http configuration.
+func (a *Agent) setupTelemetry(cfg *config.Telemetry) (*metrics.InmemSink, error) {
+
+	// Setup telemetry using an aggregate of 10 second intervals for 1 minute.
+	// Expose the metrics over stderr when there is a SIGUSR1 received.
+	inm := metrics.NewInmemSink(10*time.Second, time.Minute)
+	metrics.DefaultInmemSignal(inm)
+
+	var telConfig *config.Telemetry
+	if cfg == nil {
+		telConfig = &config.Telemetry{}
+	} else {
+		telConfig = cfg
+	}
+
+	metricsConf := metrics.DefaultConfig("nomad-autoscaler")
+	metricsConf.EnableHostname = !telConfig.DisableHostname
+	metricsConf.EnableHostnameLabel = !telConfig.DisableHostname
```

Comment:
Are these two supposed to be `telConfig.DisableHostname`?

Reply:
Nope, good spot!